src/Task/WrapperTask.php

```diff
@@ -0,0 +1,188 @@
+    }
+
+    /**
+     * Derived classes should override the getDelegate() method, and
+     * return an instance of the API class being wrapped.  When this
+     * is done, any method of the delegate is available as a method of
+     * this class.  Calling one of the delegate's methods will defer
+     * execution until the run() method is called.
+     */
+    protected function getDelegate() {
+        return null;
+    }
+
+    /**
+     * Derived classes that have more than one delegate may override
+     * getCommandList to add as many delegate commands as desired to
+     * the list of potential functions that __call() tried to find.
+     */
+    protected function getDelegateCommandList($function) {
+        return [[$this, "_$function"], [$this->getDelegate(), $function]];
+    }
+
+    /**
+     * Print progress about the commands being executed
+     */
+    protected function printTaskProgress($command, $action) {
+        $this->printTaskInfo("{$command[1]} " . json_encode($action));
+    }
+
+    /**
+     * Derived classes can override processResult to add more
+     * logic to result handling from functions. By default, it
+     * is assumed that if a function returns in int, then
+     * 0 == success, and any other value is the error code.
+     */
+    protected function processResult($function_result) {
+        if (is_int($function_result)) {
+            if ($function_result) {
+                return Result::error($this, $function_result);
+            }
+        }
+        return Result::success($this);
+    }
+
+    /**
+     * Record a function to call later.
+     */
+    protected function addToCommandStack($command, $args) {
+        $this->stack[] = array_merge([$command], $args);
+        return $this;
+    }
+
+    /**
+     * Any API function provided by the delegate that executes immediately
+     * may be handled by __call automatically.  These operations will all
+     * be deferred until this task's run() method is called.
+     */
+    public function __call($function, $args)
+    {
+        foreach ($this->getDelegateCommandList($function) as $command) {
+            if (method_exists($command[0], $command[1])) {
+                // If the function is a getter, call it now and
+                // return the result.
+                if ($this->isGetter($command[1])) {
+                    return call_user_func_array($command, $args);
+                }
+                // If the function is a setter, call it now, and
+                // return $this.
+                if ($this->isSetter($command[1])) {
+                    call_user_func_array($command, $args);
+                    return $this;
+                }
+                // Otherwise, we'll defer calling this function
+                // until run(), and return $this.
+                $this->addToCommandStack($command, $args);
+                return $this;
+            }
+        }
+
+        die("Method $function does not exist.\n");
```

Comment:
Sorry, put this in as a placeholder until I decided which exception to throw, then forgot about it. Decided to use RuntimeException, like DynamicParams does.